_posts/2017-02-29-upgrade-to-babel-7.md

```diff
@@ -0,0 +1,228 @@
+
+We highly encourage you to use a newer version of Node.js (LTS v4, LTS v6) since the previous versions are not maintained.
+See [nodejs/LTS](https://github.com/nodejs/LTS) for more information.
+
+> This just means Babel *itself* won't run on older versions of Node. It can still *output* code that runs on old Node. 
+
+## babel-preset-stage-3
+
+> Remove Stage 4 plugins from Stage 3 [#5126](https://github.com/babel/babel/pull/5126) ![high](https://img.shields.io/badge/risk%20of%20breakage%3F-high-red.svg)
+
+These plugins were moved into their yearly presets after moving to Stage 4:
+
+`babel-plugin-syntax-trailing-function-commas` (babel-preset-es2017)
+
+Example:
+
+```js
+clownPuppiesEverywhere(
+  'foo',
+  'bar', // Next parameter that's added only has to add a new line, not modify this line
+);
+```
+
+`babel-plugin-transform-async-to-generator` (babel-preset-es2017)
+
+Example:
+
+```js
+async function foo() {
+    await bar();
+}
+```
+
+`babel-plugin-transform-exponentiation-operator` (babel-preset-es2016)
+
+Example:
+
+```js
+let cubed = 2 ** 3;
+```
+
+---
+
+Instead of any yearly preset, we suggest that you use newly created `babel-preset-env` which uses the correct plugins based on your environment.
+
+Before
+
+```json
+{
+  "presets": ["es2015", "es2016", "es2017"]
+}
+```
+
+After
+
+```json
+{
+  "presets": ["env"]
+}
+```
+
+```json
+{
+  "presets": [
+    ["env", {
+      "targets": {
+        "browsers": ["last 2 versions", "safari >= 7"]
+      }
+    }]
+  ]
+}
+```
+
+See [/docs/plugins/preset-env/](/docs/plugins/preset-env/) for more information.
+
+## Spec Compliancy 
+
+> A trailing comma cannot come after a RestElement in objects [#290](https://github.com/babel/babylon/pull/290) ![medium](https://img.shields.io/badge/risk%20of%20breakage%3F-medium-yellow.svg)
+
+This is when you are using `babel-plugin-transform-object-rest-spread`
+
+Before:
+
+```js
+var { ...y, } = { a: 1};
+```
+
+This will now throw a SyntaxError.
+
+After:
+
+```js
+var { ...y } = { a: 1};
+```
+
+or:
+
+```js
+var { ...y, b } = { a: 1};
+```
+
+## babel-preset-stage-1/babel-preset-stage-2 (decorators)
+
+> [legacy-decorators](https://github.com/loganfsmyth/babel-plugin-transform-decorators-legacy) has been moved into the [transform-decorators](https://github.com/babel/babel/tree/master/packages/babel-plugin-transform-decorators) package [#5225](https://github.com/babel/babel/pull/5225) ![medium](https://img.shields.io/badge/risk%20of%20breakage%3F-medium-yellow.svg)
+
+We don't currently have a Stage 2 transform for decorators so instead of making it error, we are adding what was legacy-decorators as part of the Stage 1 preset by merging it into the transform-decorators plugin.
```

Comment:
Currently, we don't have a Stage 2 transform for decorators. Instead of making it error, we are adding legacy-decorators as part of the Stage 1 preset by merging it into the transform-decorators plugin.